Continuous.Client.Core/ContinuousEnv.MonoDevelop.cs

```diff
@@ -151,112 +163,135 @@ public override void SetTypeCode ()
+		protected override async Task<TypeDecl[]> GetTopLevelTypeDeclsAsync ()
+		{
+			var doc = IdeApp.Workbench.ActiveDocument;
+			Log ("Doc = {0}", doc);
+			if (doc == null) {
+				return new TypeDecl[0];
+			}
+
+			var ext = doc.FileName.Extension;
+
+			if (ext == ".cs") {
+				try {
+
+                    var analysisDocument = doc.GetAnalysisDocument();
+
+					var root = await analysisDocument.GetSyntaxRootAsync ();
+					var model = await analysisDocument.GetSemanticModelAsync ();
+					var typeDecls =
 						root.DescendantNodes ((arg) => true)
-							.OfType<ClassDeclarationSyntax> ()
-							.Select (t => new CSharpTypeDecl {
-								Document = new DocumentRef (doc.FileName.FullPath),
-								Declaration = t,
-								Root = root,
-								Model = model,
-							});
-					return typeDecls.ToArray ();
-
-				} catch (Exception ex) {
-					Log (ex);
-					return new TypeDecl[0];
-				}
-			}
-
-			if (ext == ".xaml") {
-				var xaml = doc.Editor.Text;
-				return new TypeDecl[] {
-					new XamlTypeDecl {
-						Document = new DocumentRef (doc.FileName.FullPath),
-						XamlText = xaml,
-					},
-				};
-			}
-
-			return new TypeDecl[0];
-		}
-
-		protected override async Task<TextLoc?> GetCursorLocationAsync ()
-		{
-			var doc = IdeApp.Workbench.ActiveDocument;
-			if (doc == null) {
-				return null;
-			}
-
-			var editLoc = doc.Editor.CaretLocation;
-			return new TextLoc (editLoc.Line, editLoc.Column);
+							.OfType<ClassDeclarationSyntax> ()
+							.Select (t => new CSharpTypeDecl {
+								Document = new DocumentRef (doc.FileName.FullPath),
+								Declaration = t,
+								Root = root,
+								Model = model,
+							});
+					return typeDecls.ToArray ();
+
+				} catch (Exception ex) {
+					Log (ex);
+					return new TypeDecl[0];
+				}
+			}
+            
+			return new TypeDecl[0];
+		}
+
+		protected override async Task<TextLoc?> GetCursorLocationAsync ()
+		{
+			var doc = IdeApp.Workbench.ActiveDocument;
+			if (doc == null) {
+				return null;
+			}
+
+            var editLoc = doc.GetContent<ITextView>().Caret.Position.BufferPosition.GetLineAndColumn1Based();
+			return new TextLoc (editLoc.line, editLoc.column);
 		}
 
 		protected override void MonitorEditorChanges ()
 		{
 			IdeApp.Workbench.ActiveDocumentChanged += BindActiveDoc;
 			BindActiveDoc (this, EventArgs.Empty);
 		}
-
+        
 		MonoDevelop.Ide.Gui.Document boundDoc = null;
+        ITextBuffer2 buffer = null;
 
 		void BindActiveDoc (object sender, EventArgs e)
 		{
 			var doc = IdeApp.Workbench.ActiveDocument;
+            
 			if (boundDoc == doc) {
 				return;
 			}
-			if (boundDoc != null) {				
-				boundDoc.DocumentParsed -= ActiveDoc_DocumentParsed;
-			}
+            
+			if (buffer != null) {
+                buffer.Changed -= ScheduleUpdate; 
+            }
+
 			boundDoc = doc;
-			if (boundDoc != null) {
-				boundDoc.DocumentParsed += ActiveDoc_DocumentParsed;
+            buffer = doc.GetContent<ITextBuffer2>();
```

Comment:
If you use `ITextBuffer` here instead of `ITextBuffer2` it will work with both the old and the new text editor.